virttest/utils_test/libvirt.py

```diff
@@ -2846,9 +2846,24 @@ def set_vm_disk(vm, params, tmp_dir=None, test=None):
         If disk_src_name is given, replace current source file
         Otherwise, use current source file with update params.
         """
-        if disk_src_name:
-            blk_source = disk_src_name
-        disk_params_src = {"source_file": blk_source}
+        if disk_type == "block":
+            src_lv_name = params.get("src_lv_name")
+            src_vg_name = params.get("src_vg_name")
+            slice_offset = params.get("slice_offset")
+            slice_size = params.get("slice_size")
```

Comment:
I think it's better to have default values , or check if they have proper values before setting them. what do you think?